flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/StreamingMonitorFunction.java

```diff
@@ -79,6 +83,16 @@ public StreamingMonitorFunction(TableLoader tableLoader, ScanContext scanContext
     this.scanContext = scanContext;
   }
 
+  @Override
+  public void open(Configuration parameters) throws Exception {
+    super.open(parameters);
+
+    final String jobId = getRuntimeContext().getJobId().toString();
+    this.workerPool = ThreadPools.newKeyedWorkerPool(jobId, "flink-worker-pool", scanContext.planParallelism());
```

Comment:
I saw this shares the same key as `IcebergFilesCommitter`, but not `FlinkInputFormat`. Trying to understand the reasons.

Reply:
I agree here. Since this is creating a different thread pool per job ID, the thread name prefix should also include the job ID to get unique names.